### DISCUSSION TITLE
Would like know the list of API

### DISCUSSION THREAD

**Discussion by charankumarpalla** [freeplane/freeplane#816] [Help] [answered]
Hi , 

i have been a freeplane user for a longtime and i would like to automate some of my task flows in freeplane through writing scripting(most probably javascript )

Not sure , but i dont find any api's docs, reference methods for this.
A little info on [javascript is seen here](https://docs.freeplane.org/attic/old-mediawiki-content/Scripting!_Other_languages.html#built-in-javascript)
i would really appreciate any such pointers on how to start writing custom scripts in Javascript

**Comment by sifran-github** [comment, marked as answer]
The api reference can be found inside Freeplane menu at: Help -> Freeplane API.

I also planned a restructure of the scripting documentation here: https://github.com/freeplane/docs/discussions/76 and i plan to implement it soon into the documentation. I would appreciate your opinion on it before implementing it. It will help us to have a place to put small examples of the API (cheat sheet for developers).

**Reply by charankumarpalla** [reply]
> freeplane/docs#76

Perfect....I see the restructuring documentation looks easier to understand
